qt/python/mantidqt/mantidqt/dialogs/errorreports/report.py

```diff
@@ -74,19 +75,32 @@ def __init__(self, parent=None, show_continue_terminate=False):
             self.requestTextBrowser.insertHtml(msg)
 
         self.input_name_line_edit.textChanged.connect(self.set_button_status)
+        self.input_email_line_edit.textChanged.connect(self.toggle_valid_email)
         self.input_email_line_edit.textChanged.connect(self.set_button_status)
         self.input_free_text.textChanged.connect(self.set_plain_text_edit_field)
         self.input_free_text.textChanged.connect(self.set_plain_text_length_lable)
         self.input_free_text.textChanged.connect(self.set_button_status)
 
+        self.email_notice_expanded_text.hide()
+        self.email_notice_button.clicked.connect(self.expand_or_hide_email_notice_text)
+
+        # https://stackabuse.com/python-validate-email-address-with-regular-expressions-regex/
+        self.email_regex = re.compile(
+            r"([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@"
+            r"([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\[[\t -Z^-~]*])"
```

Comment:
This regex works behaves in a bit of a weird way. It accepts the email address as soon as there is any character after the `@`, then flashes red every time you subsequently type a `.`.